mirrord/layer/src/lib.rs

```diff
@@ -304,6 +304,21 @@ fn layer_pre_initialization() -> Result<(), LayerError> {
         .map(|x| x.to_vec())
         .unwrap_or_default();
 
+    // SIP Patch the process' binary then re-execute it. Needed
+    // for https://github.com/metalbear-co/mirrord/issues/1529
+    #[cfg(target_os = "macos")]
+    if given_process.ends_with("dotnet") {
+        if let Ok(path) = std::env::current_exe() {
+            if let Ok(Some(binary)) =
+                mirrord_sip::sip_patch(path.to_str().unwrap(), &patch_binaries)
+            {
+                let err = exec::execvp(binary, std::env::args());
+                error!("Couldn't execute {:?}", err);
+                return Err(LayerError::ExecFailed(err));
+            }
+        }
+    }
```

Comment:
Is this the right place for this logic? Seems oddly specific, haven't we encountered any similiar case in the past?

Reply:
not that I know of, I thought of adding a new enum variant for the `LoadType` but seems overkillatm

Reply:
imho new variant would be more clear, keeping the logic in one place

Reply:
but it's kinda same sport (2 lines above)

Reply:
not pushing it, it's fine with me